src/vmm/src/devices/virtio/virtio_block/io/async_io.rs

```diff
@@ -66,13 +67,10 @@ impl<T: Debug> WrappedUserData<T> {
 }
 
 impl<T: Debug> AsyncFileEngine<T> {
-    pub fn from_file(file: File) -> Result<AsyncFileEngine<T>, AsyncIoError> {
-        log_dev_preview_warning("Async file IO", Option::None);
-
-        let completion_evt = EventFd::new(libc::EFD_NONBLOCK).map_err(AsyncIoError::EventFd)?;
-        let ring = IoUring::new(
+    fn new_ring(file: &File, completion_fd: RawFd) -> Result<IoUring, io_uring::IoUringError> {
```

Comment:
Looking at these codecov messages, it looks like we do have tests that cover these lines. Is that because the tests run on a 4.14 host?

Reply:
Looking at the changed files, I don't see codecov complaining(!)

Reply:
It is indeed weird, because these things end up being called by `virtio_block::device.rs` tests, so I think you might be right that it's about 4.14.